pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/Changed.groovy

```diff
@@ -46,11 +47,13 @@ public class Changed extends BuildCondition {
         }
         // If the current build's result isn't null (i.e., it's got a specified status), and it's different than the
         // previous build's result, we're changed.
-        else if (r.getResult() != null && !prev.getResult().equals(r.getResult())) {
+        else if ((execResult != null && !prev.getResult().equals(execResult)) ||
```

Comment:
similar here: `(execResult != null && prev.result != execResult) || r.result != prev.result //r.result should always be != null`